cmake/Modules/MetashellSystemTest.cmake

```diff
@@ -63,3 +63,17 @@ function(register_system_test TEST_TARGET_NAME)
   )
 endfunction()
 
+function(register_gcc_system_test TEST_TARGET_NAME)
+  if (NOT (WIN32 OR APPLE))
```

Comment:
OS X doesn't have g++ installed by default. There is a `g++` executable, but it's symlinked to `clang++`.

It is possible to install a true g++ though. I'll do it once this is merged.